### PR TITLE
Fix CircleCI MacOS tests

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Homebrew/homebrew-core@3f2a0d5052bef808d570f8f3ed2f087fece20fc7 updated `clang-format`. Since #7436 introduced a cache we need to fallback to overwriting the links when there is an update.
- I also found a path missing from cache: Python couldn't load the ssl module since OpenSSL is stored in `/usr/local/opt`

### Motivation

Fix CI

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -260,6 +260,7 @@ jobs:
             - /usr/local/Cellar
             - /usr/local/lib
             - /usr/local/Frameworks
+            - /usr/local/opt/openssl@1.1
```
Comment (albertvaka) [inline .circleci/config.yml:263]: Is this related?

Comment (mx-psi) [reply .circleci/config.yml:263]: No, sorry, I have updated the PR description to reflect this change too. I found it while investigating this: if you look here https://app.circleci.com/pipelines/github/DataDog/datadog-agent/14821/workflows/ef57b0b0-da4d-4954-9b7e-3aa83218e5a2/jobs/344996/parallel-runs/0/steps/0-105 you can see that Python can't load OpenSSL

Comment (mx-psi) [reply .circleci/config.yml:263]: Seems like this is not enough to fix it 😞 Since this is a necessary step for fixing it I will keep it and create a task to fix this

**Review by albertvaka — APPROVED** [review]
